services/ui-src/src/components/widgets/WidgetContainer.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+// components
+import { Flex } from "@chakra-ui/react";
+import { IconlessWidget, IconWidget, ImageWidget } from "components";
+// utils
+import { iconlessWidget, iconWidget, imageWidget } from "types";
+import { makeMediaQueryClasses } from "utils";
+
+export const widgetSelector = (widget: any, index: number) => {
+  const key = `widget-${index}`;
+  switch (widget.type) {
+    case "iconWidget":
```

Comment:
Could these cases be mapped as an enum somewhere rather than just matching by string? It could give you somewhere better to look for all the supported widget types, as opposed to digging through this switch block as it expands.